docs/repo-docs/guides/frameworks/framework-bindings.mdx

```diff
@@ -0,0 +1,81 @@
+---
+title: Framework bindings in libraries
+description: Learn how to create framework bindings in packages.
+---
+
+import { PackageManagerTabs, Tab } from '#/components/tabs';
+import { Callout } from '#/components/callout';
+
+Framework bindings in a [Library Package](/repo/docs/core-concepts/package-types#library-packages) integrate your library's code more deeply with a framework by leveraging APIs from the framework directly in the library.
+
+To do this, use the `peerDependencies` field in `package.json` of the library, which makes the framework APIs available in your library without installing it directly in the package.
+
+<Callout type="good-to-know">
+  On this page, we'll be using Next.js for examples, but the concepts below
+  apply to any framework or other dependency.
+</Callout>
+
+## Example
+
+Add a `peerDependency` to your library for the dependency that you intend to create bindings for.
+
+```json title="./packages/ui/package.json"
+{
+  "name": "@repo/ui",
+  "peerDependencies": {
+    "next": "*"
+  }
+}
+```
+
+<Callout type="good-to-know">
+  In the example above, the `peerDependency` for `next` accepts any version. You
+  may want to specify a range (for example, `">=15"`) according to your needs.
+</Callout>
+
+This will make the dependency available in your library, allowing you to write code like below. Note the `className` prop, which sets a default styling for this component in the monorepo and can be overridden in the `props` object.
```

Comment:
We should add a note that older package managers (pnpm 7, npm v4-v6, etc.) will not install peers by default. Sometimes this is worked around by _also_ adding this dep to devDeps, or enabling the right "install peers" option of your package manager.